dbg.h

```diff
@@ -81,11 +85,15 @@ std::string type_name() {
   return type.substr(prefixlen, type.size() - prefixlen - suffixlen);
 }
 
-template <>
-std::string type_name<std::string>() {
+std::string type_name(type_tag<std::string>) {
   return "std::string";
 }
 
+template <typename T>
+std::string type_name(type_tag<std::vector<T, std::allocator<T>>>) {
```

Comment:
Note how this only applies if the allocator is actually the standard one.

Reply:
:+1: